### DISCUSSION TITLE
error jupyter notebook pip install

### DISCUSSION THREAD

**Discussion by jcheemelaar** [medspacy/medspacy#174] [Q&A]
Hi!

i am trying to install medspacy with jupyter notebook with pip install / or conda install but i keep getting the same error:

It probably has to do with leveldb. I am running on a windows 10 computer at my institution. 

Collecting medspacy
  Using cached medspacy-1.0.0-py3-none-any.whl (146 kB)
Collecting spacy>=3.4.1
  Using cached spacy-3.4.3-cp39-cp39-win_amd64.whl (11.9 MB)
Collecting pysbd==0.3.4
  Using cached pysbd-0.3.4-py3-none-any.whl (71 kB)
Collecting PyRuSH>=1.0.6
  Using cached PyRuSH-1.0.8-cp39-cp39-win_amd64.whl (64 kB)
Collecting medspacy-quickumls==2.6
  Using cached medspacy_quickumls-2.6-py3-none-any.whl (31 kB)
Requirement already satisfied: jsonschema in c:\users\jch97\anaconda3\lib\site-packages (from medspacy) (4.16.0)
Collecting unqlite>=0.8.1
  Using cached unqlite-0.9.3.tar.gz (575 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting leveldb>=0.193
  Using cached leveldb-0.201.tar.gz (236 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
Note: you may need to restart the kernel to use updated packages.

error: subprocess-exited-with-error
  
  python setup.py egg_info did not run successfully.
  exit code: 1
  
  [1 lines of output]
  Don't know how to compile leveldb for Windows!
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

Encountered error while generating package metadata.

See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by burgersmoke** [comment]
Yes, I am trying to figure out how this is happening.  When we released the 1.0.0 version, a pip install was working fine on Windows and now it is not. 

For now, one workaround is to check out the repo (or download a ZIP of it) and run this from the location of your medspacy repo:

`python setup.py install`

Can you try this and see if it resolves your issue?

As a side note for anyone interested, `leveldb` was once a dependency for `medspacy_quickumls` but we removed this dependency in the last release.  So I'm trying to figure answers to the following questions.  I'm open to any ideas from users or developers of medspacy:

1. This is an older dependency for medspacy_quickumls , but on line 12 of our setup.py I already had code which should prevent medspacy_quickumls from attempting to install.  Anyone have any ideas on why this might work properly for python setup.py install and not for a pip install? (FYI: https://github.com/medspacy/medspacy/blob/master/setup.py#L12)
2. Further, even if medspacy_quickumls attempts to install, I removed leveldb as a dependency (and even as an option) with several tickets.  Is it possible I may have published medspacy while I had an older version of medspacy_quickumls in my environment?
3. Any ideas on why this started recently (past 2-3 weeks) while when we released a pip install would work fine on Windows?

**Comment by jcheemelaar** [comment]
This worked for me! Thank you so much 🙂 i have been trying to get it to work for hours. Let NLP commence.

Best
________________________________
Van: Kelly Peterson ***@***.***>
Verzonden: maandag 28 november 2022 20:31
Aan: medspacy/medspacy ***@***.***>
CC: jcheemelaar ***@***.***>; Author ***@***.***>
Onderwerp: Re: [medspacy/medspacy] error jupyter notebook pip install (Discussion #174)


Yes, I am trying to figure out how this is happening. When we released the 1.0.0 version, a pip install was working fine on Windows and now it is not.

For now, one workaround is to check out the repo (or download a ZIP of it) and run this from the location of your medspacy repo:

python setup.py install

Can you try this and see if it resolves your issue?

As a side note for anyone interested, leveldb was once a dependency for medspacy_quickumls but we removed this dependency in the last release. So I'm trying to figure answers to the following questions. I'm open to any ideas from users or developers of medspacy:

  1.  This is an older dependency for medspacy_quickumls , but on line 12 of our setup.py I already had code which should prevent medspacy_quickumls from attempting to install. Anyone have any ideas on why this might work properly for python setup.py install and not for a pip install? (FYI: https://github.com/medspacy/medspacy/blob/master/setup.py#L12)
  2.  Further, even if medspacy_quickumls attempts to install, I removed leveldb as a dependency (and even as an option) with several tickets. Is it possible I may have published medspacy while I had an older version of medspacy_quickumls in my environment?
  3.  Any ideas on why this started recently (past 2-3 weeks) while when we released a pip install would work fine on Windows?

**Comment by burgersmoke** [comment]
Fantastic.  Let the NLP flow! :)

**Comment by burgersmoke** [comment]
For anyone who wants to test this early, could you try this on Windows and let me know if this works for you?

`pip install medspacy>=1.0.1`

**Comment by burgersmoke** [comment]
OK, this appears to be working now with 1.0.1 which was just merged into master.
